crowbar_framework/app/models/cinder_service.rb

```diff
@@ -205,6 +205,20 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     # the VIP of the cluster to be setup
     allocate_virtual_ips_for_any_cluster_in_networks(controller_elements, vip_networks)
 
+    # Generate secrets uuid for libvirt rbd backend
+    proposal = ProposalObject.find_data_bag_item "crowbar/bc-cinder-default"
```

Comment:
Yes, I agree. I was thinking also about moving this part to javascript and I'm sure that it will be the best solution.